packages/@react-aria/textfield/src/useFormattedTextField.ts

```diff
@@ -77,6 +77,9 @@ export function useFormattedTextField(props: AriaTextFieldProps, state: Formatte
             ? input.value.slice(0, input.selectionStart - 1) + input.value.slice(input.selectionStart)
             : input.value.slice(0, input.selectionStart) + input.value.slice(input.selectionEnd);
           break;
+        case 'deleteSoftLineBackward':
```

Comment:
Looks like Safari treats CMD+Delete as `deleteHardLineBackward` so you'll need to add that here as well.